### PR TITLE
Supporting Azure PL in CLI

### PR DESCRIPTION
**Pull request by nchen99**
Release Notes
-------------

New Features

- Support creating Azure Private Link attachments in `confluent network private-link attachment create`
- Output the associated Azure Private Link Service alias and ID in `confluent network private-link attachment describe` and `confluent network private-link attachment list`
- Support creating Azure Private Link attachment connections with `confluent network private-link attachment connection create`
- Output the associated Azure Private Link Service alias and ID in `confluent network private-link attachment connection describe` and `confluent network private-link attachment connection list`

Checklist
---------
- [ ] Leave this box unchecked if features are not yet available in production

What
----
Adding support for Azure Private Link connections.

References
----------

Test & Review
-------------

### CODE REVIEW

**Comment by brianstrauch** [comment]
> Privatelink Attachment Create supports creating Azure PLs
Privatelink Attachment list and describe supports describing Azure specific fields

These release notes are user-facing, so please type out the full command name and format with backticks

**Comment by nchen99** [comment]
> These release notes are user-facing, so please type out the full command name and format with backticks

yes for sure! I will format everything properly for the final PR. Creating draft to double check that these are all the fields we need to support azure pl connections / in case I missed anything just in the commands themselves

File: internal/network/command_private_link_attachment_connection_create.go
Comment (linouk23) [reply internal/network/command_private_link_attachment_connection_create.go:30]: I'll defer to @brianstrauch / @ethanruhe for CLI specific questions.

File: internal/network/command_private_link_attachment_connection_create.go
Comment (nchen99) [reply internal/network/command_private_link_attachment_connection_create.go:22]: Good point, I think it would be one of 
- Private Link
- PrivateLink

Just not sure if the standard is to have the space or not, I'll follow up in my team

Comment (syoung-cflt) [reply internal/network/command_private_link_attachment_connection_create.go:22]: I think AWS uses PrivateLink and Azure uses Private Link.

Comment (syoung-cflt) [reply internal/network/command_private_link_attachment_connection_create.go:22]: I think we use both "Private Link" and "Private Link` in our doc 🤔

Comment (nchen99) [reply internal/network/command_private_link_attachment_connection_create.go]: Team decided that it should be "Private Link", made the changes accordingly in the files!

File: test/fixtures/output/network/private-link/attachment/connection/describe-aws-json.golden
Comment (dhathri) [inline test/fixtures/output/network/private-link/attachment/connection/describe-aws-json.golden:5]: why do we need this change? `private-link-attachment-id` should be of the form `platt-xxx` unless this is something different 🤔

**Review by dhathri — COMMENTED** [review]
In all tests, `private-link-attachment-id` should be of the form `platt-xxxx`. `private-link-attachment` is a different resource than `private-link-attachment-connection` and `plattc-xxx` references `platt-xxx` in the spec. Let's please fix the tests. Otherwise looks good.

File: internal/network/command_private_link_attachment.go
Comment (brianstrauch) [inline internal/network/command_private_link_attachment.go:94]: Cool, didn't know we generated this function

File: internal/network/command_private_link_attachment_connection_create.go
Comment (brianstrauch) [inline internal/network/command_private_link_attachment_connection_create.go:22]:
```suggestion
				Text: `Create a Private Link attachment connection named "aws-private-link-attachment-connection".`,
```

File: internal/network/command_private_link_attachment_connection.go
Comment (brianstrauch) [inline internal/network/command_private_link_attachment_connection.go:22]:
```suggestion
	AzurePrivateEndpointResourceId string `human:"Azure Private Endpoint Resource ID,omitempty" serialized:"azure_private_endpoint_resource_id,omitempty"`
```

**Review by dhathri — DISMISSED** [review]
LGTM. 
I'm not sure what is the process for cIi release. We will need this to be released as part of Azure E-sku GA which is scheduled for 3/31..  And not before that..
@brianstrauch Please let us know if we need to hold off merging if we don't want this released until later.

**Comment by brianstrauch** [comment]
> LGTM. I'm not sure what is the process for cIi release. We will need this to be released as part of Azure E-sku GA which is scheduled for 3/31.. And not before that.. @brianstrauch Please let us know if we need to hold off merging if we don't want this released until later.

If you need to release exactly on 3/31 we need to gate this feature with a feature flag since 3/31 is a Sunday and the CLI team won't be available to run a release. Otherwise, let's wait to merge it until after 3/31 and include it in our regularly scheduled release on Thursday 4/4.

File: internal/network/command_private_link_attachment_connection_create.go
Comment (brianstrauch) [reply internal/network/command_private_link_attachment_connection_create.go:30]: Let's remove this comment if this discussion is resolved.

File: test/fixtures/output/network/private-link/attachment/describe-azure.golden
Comment (brianstrauch) [inline test/fixtures/output/network/private-link/attachment/describe-azure.golden:3]: These are so similarly named it's confusing which one is the ID and which one is the name... maybe rename to the following:
```suggestion
| ID                             | platt-azure                      |
| Name                           | my-azure-private-link-attachment |
```

**Comment by brianstrauch** [comment]
Updated the release notes to match the formatting of our existing release notes, please give them a read through @nchen99!

Trying to understand this PR: Do you think these two features are redundant? (Do both commands necessarily both need to print the same information?)
* Output the associated Azure Private Link Service alias and ID in `confluent network private-link attachment describe` and `confluent network private-link attachment list`
* Output the associated Azure Private Link Service alias and ID in `confluent network private-link attachment connection describe` and `confluent network private-link attachment connection list`

**Comment by nchen99** [comment]
> Updated the release notes to match the formatting of our existing release notes, please give them a read through @nchen99!
> 
> Trying to understand this PR: Do you think these two features are redundant? (Do both commands necessarily both need to print the same information?)
> 
> * Output the associated Azure Private Link Service alias and ID in `confluent network private-link attachment describe` and `confluent network private-link attachment list`
> * Output the associated Azure Private Link Service alias and ID in `confluent network private-link attachment connection describe` and `confluent network private-link attachment connection list`

Thank you so much for updating the release notes! They look great!

Essentially there are two different objects here:
- private link attachment
- private link attachment connection

One is the private link attachment itself, and the other is the connection associated with the private link attachment. This happens with AWS too, as both commands will print out the VPC endpoint ID. It's necessary to see what Azure resources are associated with this specific attachment or attachment connection. Even though it's technically the same piece of information, that data will be different as they are associated with different objects.

**Review by brianstrauch — APPROVED** [review]

**Comment by dhathri** [comment]
> > LGTM. I'm not sure what is the process for cIi release. We will need this to be released as part of Azure E-sku GA which is scheduled for 3/31.. And not before that.. @brianstrauch Please let us know if we need to hold off merging if we don't want this released until later.
> 
> If you need to release exactly on 3/31 we need to gate this feature with a feature flag since 3/31 is a Sunday and the CLI team won't be available to run a release. Otherwise, let's wait to merge it until after 3/31 and include it in our regularly scheduled release on Thursday 4/4.

cc: @TylerPollak I think we can just merge this after 3/31 and release it as part of scheduled CLI release on 4/4. PLMK if that sounds good.

**Comment by brianstrauch** [comment]
> What date is the release prior to 4/4?

The Thursday before, so 3/28

> Between 3/31 and 4/4 would the only option be to gate behind an LD flag?

No, we'd be happy to make an exception and release on 4/1, just let us know!

File: internal/network/command_private_link_attachment_connection.go
Comment (syoung-cflt) [inline internal/network/command_private_link_attachment_connection.go:78]: do we still need this clause?
`if connection.Status.Cloud != nil && connection.Status.Cloud.NetworkingV1AwsPrivateLinkAttachmentConnectionStatus != nil`

File: test/test-server/networking_handlers.go
Comment (syoung-cflt) [inline test/test-server/networking_handlers.go:1261]: non blocking but it would be great if we can have real examples here and check how they render in the output

File: test/network_test.go
Comment (syoung-cflt) [inline test/network_test.go:406]: we should add more test cases if we expect different output for different states

Comment (nchen99) [reply test/network_test.go:406]: I think the other test cases are common between aws and azure, the only thing specific to azure would be to output the alias and the id which would be covered by calling describe for an azure platt

File: internal/network/command_private_link_attachment_connection.go
Comment (nchen99) [reply internal/network/command_private_link_attachment_connection.go]: will remove!

File: test/test-server/networking_handlers.go
Comment (brianstrauch) [inline test/test-server/networking_handlers.go:1269]:
```suggestion
```

**Comment by TylerPollak** [comment]
> > What date is the release prior to 4/4?
> 
> The Thursday before, so 3/28
> 
> > Between 3/31 and 4/4 would the only option be to gate behind an LD flag?
> 
> No, we'd be happy to make an exception and release on 4/1, just let us know!

Thanks @brianstrauch that helps. Let's target the 3/28 release as our launch will likely be +/- a few days from 3/31 (Sunday). I'd rather not juggle LD flags on the GA product

**Review by brianstrauch — DISMISSED** [review]

**Review by brianstrauch — DISMISSED** [review]

**Comment by brianstrauch** [comment]
Postponing for next week's release

**Review by dhathri — APPROVED** [review]

**Pull request merged**
